### PR TITLE
Exit from `bundle` with Bundler's intended exit code

### PR DESCRIPTION
**Pull request by dmikurube**
Same as #1599 for `embulk bundle`.

### CODE REVIEW

File: embulk-core/src/main/java/org/embulk/cli/EmbulkRun.java [hunk 1]
```diff
@@ -235,20 +235,39 @@ private int runBundler(final List<String> arguments, final Path path, final Embu
         localJRubyContainer.runScriptlet("require 'bundler/friendly_errors'");
         localJRubyContainer.runScriptlet("require 'bundler/cli'");
 
+        localJRubyContainer.runScriptlet("__internal_exit_status__ = 0");
         localJRubyContainer.put("__internal_argv_java__", arguments);
         if (path == null) {
-            localJRubyContainer.runScriptlet("Bundler.with_friendly_errors { Bundler::CLI.start(Array.new(__internal_argv_java__), debug: true) }");
+            localJRubyContainer.runScriptlet(
+                    "begin;"
+                    + "  Bundler.with_friendly_errors { Bundler::CLI.start(Array.new(__internal_argv_java__), debug: true) }"
```
Comment (hiroyuki-sato) [inline embulk-core/src/main/java/org/embulk/cli/EmbulkRun.java:243]: missing `;`? 

```
embulk-0.10.48-SNAPSHOT.jar bundle hoge

...
Exception in thread "main" org.jruby.embed.ParseFailedException: (SyntaxError) <script>:1: syntax error, unexpected =>
... true) }  rescue SystemExit => __internal_system_exit__;    ...
                                                                                                                           ^~
	at org.jruby.embed.internal.EmbedRubyRuntimeAdapterImpl.runParser(EmbedRubyRuntimeAdapterImpl.java:199)
	at org.jruby.embed.internal.EmbedRubyRuntimeAdapterImpl.parse(EmbedRubyRuntimeAdapterImpl.java:104)
	at org.jruby.embed.ScriptingContainer.parse(ScriptingContainer.java:1228)
	at org.jruby.embed.ScriptingContainer.runScriptlet(ScriptingContainer.java:1288)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.embulk.jruby.ScriptingContainerDelegateImpl.runScriptlet(ScriptingContainerDelegateImpl.java:841)
	at org.embulk.jruby.LazyScriptingContainerDelegate.runScriptlet(LazyScriptingContainerDelegate.java:305)
	at org.embulk.cli.EmbulkRun.runBundler(EmbulkRun.java:241)
	at org.embulk.cli.EmbulkRun.runInternal(EmbulkRun.java:64)
	at org.embulk.cli.EmbulkRun.run(EmbulkRun.java:24)
	at org.embulk.cli.Main.main(Main.java:53)
Caused by: org.jruby.exceptions.SyntaxError: (SyntaxError) <script>:1: syntax error, unexpected =>
... true) }  rescue SystemExit => __internal_system_exit__;    ...
```

**Review by hiroyuki-sato — CHANGES_REQUESTED** [review]
This change reports a syntax error. Could you take a look?

File: embulk-core/src/main/java/org/embulk/cli/EmbulkRun.java (hunk 1 above)
Comment (dmikurube) [reply embulk-core/src/main/java/org/embulk/cli/EmbulkRun.java]: Thanks for your good catch! Fixed, with some more updates.

**Review by hiroyuki-sato — APPROVED** [review]
LGTM👍

One comment. `embulk gem` command outputs `Exiting RubyGems with exit_code 1`. Does it output by RubyGems, not Embulk?
(Bundler command (This PR change) doesn't output those lines.)

```
embulk-0.10.48-SNAPSHOT.jar gem hogehoge

...
ERROR:  While executing gem ... (Gem::CommandLineError)
    Unknown command hogehoge
Exiting RubyGems with exit_code 1
```

<hr>


### mkbundle with error.

```
embulk-0.10.48-SNAPSHOT.jar mkbundle /dev/null

... 
'/dev/null' already exists.
palolovalley:tmp hsato$ echo $?
1
```


### mkbundle without error.

```
embulk-0.10.48-SNAPSHOT.jar mkbundle hoge

...

2023-04-11 21:29:00.339 +0900 [DEBUG] (main): Embulk system property "default_guess_plugin" is set to: "gzip,bzip2,json,csv"
Initializing hoge...
  Creating Gemfile
  Creating .bundle/config
  Creating embulk/input/example.rb
  Creating embulk/output/example.rb
  Creating embulk/filter/example.rb
2023-04-11 21:29:00.416 +0900 [INFO] (main): Loaded JRuby runtime 9.3.6.0
2023-04-11 21:29:01.611 +0900 [INFO] (main): Environment variable "GEM_HOME" is not set. Setting "GEM_HOME" to "/Users/hsato/.embulk/lib/gems" from Embulk system property "gem_home" for the "bundle" command.
Fetching gem metadata from https://rubygems.org/........
Resolving dependencies...
Using bundler 2.2.29
Fetching msgpack 1.7.0 (java)
Installing msgpack 1.7.0 (java)
Fetching embulk 0.10.47 (java)
Installing embulk 0.10.47 (java)
Bundle complete! 1 Gemfile dependency, 3 gems now installed.
Bundled gems are installed into `./.`
palolovalley:tmp hsato$ echo $?
0
```

### bundle with error.

```
embulk-0.10.48-SNAPSHOT.jar bundle hoge

...
Users/hsato/.embulk/lib/gems" from Embulk system property "gem_home" for the "bundle" command.
Could not find command "hoge".
palolovalley:hoge hsato$ echo $?
15
```

### bundle without error.

```
embulk-0.10.48-SNAPSHOT.jar bundle

..
Bundle complete! 1 Gemfile dependency, 3 gems now installed.
Bundled gems are installed into `./.`
palolovalley:hoge hsato$ echo $?
0
```

**Pull request merged**